src/platform/interpreter/types.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Uri } from 'vscode';
+import { IDisposable, InterpreterUri, Resource } from '../../platform/common/types';
+import { PythonEnvironment } from '../pythonEnvironments/info';
+
+export const IReservedPythonNamedProvider = Symbol('IReservedPythonNamedProvider');
+export interface IReservedPythonNamedProvider extends IDisposable {
+    getUriOverridingReservedPythonNames(cwd: Uri): Promise<{ uri: Uri; type: 'file' | '__init__' }[]>;
+    isReserved(uri: Uri): Promise<boolean>;
+    /**
+     * Keeps track of a Uri as a file that should be ignored from all warnings related to reserved names.
+     */
+    addToIgnoreList(uri: Uri): Promise<void>;
+}
+
+export const IInterpreterPackages = Symbol('IInterpreterPackages');
+export interface IInterpreterPackages {
```

Comment:
Moved from telemetry folder

Reply:
I need to reference this from the new code, hence moved this from telemtry into Interrpeter & ensured telemetry doesnt depend on this.